### PR TITLE
Salesforce lightning compat

### PR DESCRIPTION
**Pull request by owenpearson**

### CODE REVIEW

File: src/common/lib/util/logger.ts [hunk 1]
```diff
@@ -44,7 +47,9 @@ const getDefaultLoggers = (): [Function, Function] => {
   /* Can't just check for console && console.log; fails in IE <=9 */
   if (
     (typeof Window === 'undefined' && typeof WorkerGlobalScope === 'undefined') /* node */ ||
-    (global.console && global.console.log && typeof global.console.log.apply === 'function') /* sensible browsers */
+    (globalOrWindow.console &&
+      globalOrWindow.console.log &&
+      typeof globalOrWindow.console.log.apply === 'function') /* sensible browsers */
```
Comment (Peter-Maguire) [inline src/common/lib/util/logger.ts:52]: Could this not be `typeof globalOrWindow.console?.log?.apply === 'function`

File: src/common/lib/util/logger.ts [hunk 2]
```diff
@@ -54,7 +59,7 @@ const getDefaultLoggers = (): [Function, Function] => {
           console.warn.apply(console, args);
         }
       : consoleLogger;
-  } else if (global.console && (global.console.log as unknown)) {
+  } else if (globalOrWindow.console && (globalOrWindow.console.log as unknown)) {
```
Comment (Peter-Maguire) [inline src/common/lib/util/logger.ts:62]: Same here

File: src/common/lib/util/logger.ts (hunk 1 above)
Comment (owenpearson) [reply src/common/lib/util/logger.ts]: Good idea, done in [8e9ffac](https://github.com/ably/ably-js/pull/993/commits/8e9ffac1cbfc09d0797d8e0ad9575536bd2ea09d)

File: src/common/lib/util/logger.ts (hunk 2 above)
Comment (owenpearson) [reply src/common/lib/util/logger.ts]: [8e9ffac](https://github.com/ably/ably-js/pull/993/commits/8e9ffac1cbfc09d0797d8e0ad9575536bd2ea09d)

**Review by Peter-Maguire — APPROVED** [review]

**Pull request merged**